packages/ui5-middleware-onelogin/src/cookieGetter.ts

```diff
@@ -195,7 +256,15 @@ export default class CookieGetter {
 				} catch (oError) {
 					//This error is fine, it's not locating the No button specifically for Azure
 				}
-			} else {
+				// Certificate login without certificate configuration
+			} else if (isUseCertificateEnabled && hasCertificateConfig) {
+				// Full certificate login with provided configuration
+				log.info("Login with certificate configuration. Waiting for page to load...");
+				await page.goto(attr.url, { waitUntil: "networkidle" });
+				// Add more robust certificate handling here if needed
+			} else if (isUseCertificateEnabled && !hasCertificateConfig) {
```

Comment:
Not sure if that counts as nit, however maybe you can reuse the `useClientCertificates` variable from above
```ts
const useClientCertificates = isUseCertificateEnabled && hasCertificateConfig;
```

And package the condition of line 265 into its own variable in the same place the `useClientCertificates` was defined.

Makes it a bit more readable / less confusing as the condition being checked inside the tree is the same (for 260) and follows the same pattern (for 265).

Reply:
i just removed `useClientCertificates`, is that better?

Reply:
https://github.com/ui5-community/ui5-ecosystem-showcase/pull/1079/commits/a0f688ee7f044ca54b35c070be6d3464f25a3b7d

Reply:
Sure that works! I would've done it the other way around to have less `&&` checks going on and thereby make the IFs a tad more "readable" (the check is also used in many places). But readability lays in the eye of the beholder, right? As long as it is coherent I'm fine with it. :)

Reply:
i know what you mean, in my opinion you can see in the checks that always the config options need to be set, avoid the need to look at the definiton of another bool, i know its more text but easier to read? makes that sense?

Reply:
Yeah it does and as long as the checks aren't getting ridiculously long it is fine like that. Hence, resolved (for me). :)

Just different coding styles I think. And I would make the argument that JS is smart enough to _not_ always re-evaluate the same condition(?) not like it would matter in this case but it could be an argument. ^^